x/group/spec/03_messages.md

```diff
@@ -6,134 +6,147 @@ order: 3
+It's expected to fail if metadata length is greater than `MaxMetadataLen` config.
 
 ## Msg/WithdrawProposal
 
-A proposal can be withdrawn using `MsgWithdrawProposal` which has a `address` (can be either proposer or policy admin) and a `proposal_id` (which has to be withdrawn).
+A proposal can be withdrawn using `MsgWithdrawProposal` which has an `address` (can be either a proposer or the group policy admin) and a `proposal_id` (which has to be withdrawn).
 
-+++ https://github.com/cosmos/cosmos-sdk/blob/f2d6f0e4bb1a9bd7f7ae3cdc4702c9d3d1fc0329/proto/cosmos/group/v1beta1/tx.proto#L251-L258
++++ https://github.com/cosmos/cosmos-sdk/blob/v0.46.0-beta2/proto/cosmos/group/v1/tx.proto#L303-L310
 
-It's expecting to fail if:
+It's expected to fail if:
 
-* the signer is neither policy address nor proposer of the proposal.
-* the proposal is already closed or aborted.
+- the signer is neither the group policy address nor proposer of the proposal.
+- the proposal is already closed or aborted.
 
 ## Msg/Vote
 
-A new vote can be created with the `MsgVote`, given a proposal id, a voter address, a choice (yes, no, veto or abstain) and some optional metadata bytes.
+A new vote can be created with the `MsgVote`, given a proposal id, a voter address, a choice (yes, no, veto or abstain) and some optional metadata.
 An optional `Exec` value can be provided to try to execute the proposal immediately after voting.
 
-+++ https://github.com/cosmos/cosmos-sdk/blob/9aef070625e9676d7c0acb212c17ae9dba3c32dc/proto/cosmos/group/v1beta1/tx.proto#L248-L265
++++ https://github.com/cosmos/cosmos-sdk/blob/v0.46.0-beta2/proto/cosmos/group/v1/tx.proto#L315-L333
+
+It's expected to fail if:
 
-It's expecting to fail if metadata length is greater than `MaxMetadataLen` config.
+- metadata length is greater than `MaxMetadataLen` config.
+- the proposal is not in voting period anymore.
 
 ## Msg/Exec
 
 A proposal can be executed with the `MsgExec`.
 
-+++ https://github.com/cosmos/cosmos-sdk/blob/9aef070625e9676d7c0acb212c17ae9dba3c32dc/proto/cosmos/group/v1beta1/tx.proto#L270-L278
++++ https://github.com/cosmos/cosmos-sdk/blob/v0.46.0-beta2/proto/cosmos/group/v1/tx.proto#L338-L347
 
 The messages that are part of this proposal won't be executed if:
 
-* the proposal has not been accepted.
-* the proposal status is not closed.
-* the proposal has already been successfully executed.
+- the proposal has not been accepted by the group policy.
+- the proposal has already been successfully executed.
 
 ## Msg/LeaveGroup
 
 The `MsgLeaveGroup` allows group member to leave a group.
 
-+++ https://github.com/cosmos/cosmos-sdk/blob/a635fd78663d04c5de23f4d032e5a3abea1b005a/proto/cosmos/group/v1beta1/tx.proto#L352-L361
++++ https://github.com/cosmos/cosmos-sdk/blob/v0.46.0-beta2/proto/cosmos/group/v1/tx.proto#L352-L361
 
-It's expecting to fail if:
+It's expected to fail if:
 
-* the group member is not part of the group.
```

Comment:
I believe it'll also fail if `validateDecisionPolicies` against the updated group does, we should probably mention that in `Msg/UpdateGroupMembers` too